src/Cake.Core/IO/IDirectory.cs

```diff
@@ -14,6 +14,12 @@ public interface IDirectory
         DirectoryPath Path { get; }
 
         /// <summary>
+        /// Gets the name of the directory.
+        /// </summary>
+        /// <value>The directory's name.</value>
+        string Name { get; }
```

Comment:
Yes. But in that case I would prefer to add a `Name` property to the `FilePath` and `DirectoryPath` (which could be shorthands for `GetDirectoryName` and `GetFilename`) since this information belongs with the path.